src/main/java/no/nav/folketrygdloven/beregningsgrunnlag/arbeidstaker/SjekkAvvikSammenligningsgrunnlagMotAvviksgrense.java

```diff
@@ -0,0 +1,53 @@
+package no.nav.folketrygdloven.beregningsgrunnlag.arbeidstaker;
+
+import java.math.BigDecimal;
+
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.AktivitetStatus;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.resultat.BeregningsgrunnlagPeriode;
+import no.nav.folketrygdloven.beregningsgrunnlag.regelmodell.resultat.BeregningsgrunnlagPrStatus;
+import no.nav.fpsak.nare.doc.RuleDocumentation;
+import no.nav.fpsak.nare.evaluation.Evaluation;
+import no.nav.fpsak.nare.specification.LeafSpecification;
+
+@RuleDocumentation(SjekkAvvikSammenligningsgrunnlagMotAvviksgrense.ID)
+public class SjekkAvvikSammenligningsgrunnlagMotAvviksgrense extends LeafSpecification<BeregningsgrunnlagPeriode> {
+
+    static final String ID = "FP_BR 26.1";
+    static final String BESKRIVELSE = "Har rapportert inntekt avvik mot sammenligningsgrunnlag > 25%?";
+
+    public SjekkAvvikSammenligningsgrunnlagMotAvviksgrense() {
+        super(ID, BESKRIVELSE);
+    }
+
+    @Override
+    public Evaluation evaluate(BeregningsgrunnlagPeriode grunnlag) {
+        if(erYtelseOmsorgspenger(grunnlag)) {
+            return direkteUtbetalingTilBrukerOgAvvik(grunnlag) ? ja() : nei();
+        }
+        return erAvvikStørreEnn25Prosent(grunnlag) ? ja() : nei();
+    }
+
+    private boolean erAvvikStørreEnn25Prosent(BeregningsgrunnlagPeriode grunnlag){
+        if(grunnlag.isSplitteATFLToggleErPå()){
+            return grunnlag.getBeregningsgrunnlag().getSammenligningsGrunnlagPrAktivitetstatus(AktivitetStatus.AT).getAvvikProsent().compareTo(grunnlag.getAvviksgrenseProsent()) > 0;
+        }
+        return grunnlag.getBeregningsgrunnlag().getSammenligningsGrunnlag().getAvvikProsent().compareTo(grunnlag.getAvviksgrenseProsent()) > 0;
+    }
+
+    private boolean direkteUtbetalingTilBrukerOgAvvik(BeregningsgrunnlagPeriode grunnlag){
+        return girDirekteUtbetalingTilBruker(grunnlag) && erAvvikStørreEnn25Prosent(grunnlag);
+    }
+
+    private boolean erYtelseOmsorgspenger(BeregningsgrunnlagPeriode grunnlag){
```

Comment:
Det er litt forvirrende at metoden heiter erYtelseOmsorgspenger og at den returnerer skalSjekkeRefusjonFørAvviksvurdering. For ein person som ikkje kjenner til omsorspenger gir ikkje dette meining. Kan du enten endre navn på felt-verdien på grunnlaget eller navnet på denne metoden slik at den spør på det samme?

Reply:
Har oppdatert